### PR TITLE
ceph-fuse: kill clang analyzer warnings

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

File: src/mds/Server.cc [hunk 1]
```diff
@@ -1433,7 +1433,7 @@ void Server::reply_client_request(MDRequestRef& mdr, MClientReply *reply)
     req->get_connection()->send_message(reply);
   }
 
-  if (req->is_queued_for_replay() &&
+  if (req->is_queued_for_replay() && reply &&
```
Comment (batrick) [inline src/mds/Server.cc]: Is there a way to convert this into an `assert(reply)` that makes clang happy?

Comment (tchaikov) [reply src/mds/Server.cc]: sure, as long as we are sure that `reply` is always valid here.

Comment (tchaikov) [reply src/mds/Server.cc]: hmm, after a second thought, i am not qualified to do the conversion as i don't understand mds enough to figure out where "here" is. will drop this commit.

**Comment by tchaikov** [comment]
@batrick @ukernel could you help review this commit? we can get back to the `mds/Server.cc` change later on, but this one is relatively no-brainer, i think.

**Pull request merged**